pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -60,41 +87,100 @@ func (m *cgroupManagerImpl) Destroy(cgroupConfig *CgroupConfig) error {
 	return nil
 }
 
+type subsystem interface {
+	// Name returns the name of the subsystem.
+	Name() string
+	// Set the cgroup represented by cgroup.
+	Set(path string, cgroup *libcontainerconfigs.Cgroup) error
+}
+
+// Cgroup subsystems we currently support
+var supportedSubsystems []subsystem = []subsystem{
+	&cgroupfs.MemoryGroup{},
+	&cgroupfs.CpuGroup{},
+}
+
+func setSupportedSubsytems(cgroupConfig *libcontainerconfigs.Cgroup) error {
+	for _, sys := range supportedSubsystems {
+		if _, ok := cgroupConfig.Paths[sys.Name()]; !ok {
+			return fmt.Errorf("Failed to find subsytem mount for subsytem %v", sys.Name())
+		}
+		if err := sys.Set(cgroupConfig.Paths[sys.Name()], cgroupConfig); err != nil {
+			return fmt.Errorf("Failed to set config for subsystem %v: %v", sys.Name, err)
+		}
+	}
+	if cgroupConfig.Paths["cpu"] != "" {
+		if err := cgroupfs.CheckCpushares(cgroupConfig.Paths["cpu"], cgroupConfig.Resources.CpuShares); err != nil {
```

Comment:
To be more specific we just check if the CpuShares are set to an acceptable value
ie >=2 and <some limit